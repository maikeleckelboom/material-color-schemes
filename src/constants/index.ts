export {
  DEFAULT_QUANTIZE_MAX_COLORS,
  DEFAULT_PALETTE_TONES,
  DEFAULT_SCORE_OPTIONS,
} from './defaults.ts';
export { COLOR_SCHEME_KEYS } from './color-scheme-keys.ts';
