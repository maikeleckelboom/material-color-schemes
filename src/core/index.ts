// Configuration and settings
export {
  Variant,
  ContrastLevel,
  getClosestContrastLevel,
  listContrastLevels,
  listVariants,
  type ConfigOption,
} from './config';

// Theme and scheme creation
export { createScheme } from './scheme';
export { createTheme } from './theme';
export { createColorScheme } from './color-scheme.ts';

// conversion.ts – For color-space transformations
export {
  convertToArgb,
  convertToHex,
  convertToHct,
  convertToLab,
  convertToRgb,
  convertToLstar,
} from './conversion.ts';

// formatting.ts – For string/UI representations
export { formatColorName, formatCssVarName, formatCssVarMapping } from './formatting.ts';

export { harmonize, blendHue, blendCam } from './blend.ts';
export { createCustomColor } from './color.ts';
export { quantizeColors } from './quantize.ts';

// Palette generation and management
export { createPalette, mapPaletteTones, extractPaletteColors } from './palette.ts';

// Contrast and accessibility
export { getContrastRatioOfTones, getContrastColor, isContrasting } from './contrast.ts';
export { isDisliked, fixIfDisliked } from './dislike.ts';

// Analysis and recommendations
export { scoreColorToCount, type ScoreOptions } from './scoring.ts';
