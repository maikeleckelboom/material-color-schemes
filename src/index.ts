export type * from './types';
export * from './core';
export * from './constants';
