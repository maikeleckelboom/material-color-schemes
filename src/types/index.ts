export type * from './theme.ts'
export type * from './scheme.ts'