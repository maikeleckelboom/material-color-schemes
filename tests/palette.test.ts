import { describe, expect, it } from 'vitest';
import { TonalPalette } from '@material/material-color-utilities';
import { createPalette, DEFAULT_PALETTE_TONES, mapPaletteTones } from '../src';

describe('Tonal Palette Creation', () => {
  describe('createPalette', () => {
    const hexPalette = createPalette('#40A127');
    const argbPalette = createPalette(0xFF40A127);

    it('creates a TonalPalette from valid color inputs (hex/ARGB)', () => {
      expect(hexPalette).toBeInstanceOf(TonalPalette);
      expect(argbPalette).toBeInstanceOf(TonalPalette);
    });


    it('produces equivalent palettes for hex and ARGB inputs', () => {
      expect(hexPalette.tone(40)).toEqual(argbPalette.tone(40));
    })
  });

  describe('mapPaletteTones', () => {
    const testColor = '#40A127';
    const testTones = [0, 50, 100];

    it('generates tone maps for default and custom tones with order preservation', () => {
      // Default tones
      const defaultMap = mapPaletteTones(testColor, [...DEFAULT_PALETTE_TONES]);
      expect(Object.keys(defaultMap)).toEqual(DEFAULT_PALETTE_TONES.map(String));

      // Custom tones with edge cases
      const customMap = mapPaletteTones(testColor, testTones);
      expect(Object.keys(customMap)).toEqual(testTones.map(String));
    });

    it('uses palette.tone() for each specified tone', () => {
      const palette = createPalette(testColor);
      const tones = [20, 40, 60];
      const toneMap = mapPaletteTones(palette, tones);

      tones.forEach(tone => {
        expect(toneMap[tone]).toBe(palette.tone(tone));
      });
    });

    it('handles both color formats consistently', () => {
      const hexMap = mapPaletteTones('#40A127');
      const argbMap = mapPaletteTones(0xFF40A127);

      expect(hexMap).toEqual(argbMap);
    });

    it('does not mutate input tones array', () => {
      const originalTones = [10, 20];
      const copiedTones = [...originalTones];
      mapPaletteTones(testColor, originalTones);

      expect(originalTones).toEqual(copiedTones);
    });

    it('matches default tones snapshot', () => {
      expect(mapPaletteTones(testColor)).toMatchSnapshot();
    });
  });
});