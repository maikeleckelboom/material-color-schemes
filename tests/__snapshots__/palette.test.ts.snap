// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tonal Palette Creation > mapPaletteTones > matches default tones snapshot 1`] = `
{
  "0": 4278190080,
  "10": 4278395136,
  "100": 4294967295,
  "20": 4278728960,
  "30": 4279194368,
  "40": 4279725568,
  "50": 4280715532,
  "60": 4282688810,
  "70": 4284465220,
  "80": 4286242140,
  "90": 4288019061,
  "95": 4291559349,
  "99": 4294442989,
}
`;
